### ISSUE TITLE
Use Debian way to choose default java version

### ISSUE THREAD

**Issue by mikefaille** [GELOG/docker-ubuntu-hadoop#20]
To choose default java version, I think we must use Debian way to enforce system validation :  https://askubuntu.com/questions/315646/update-java-alternatives-vs-update-alternatives-config-java

Reason ? It will comply with further system updates.

My reference is this commit: https://github.com/GELOG/docker-ubuntu-hadoop/pull/15/files

cc : @codingtony

**Comment by davidonlaptop** [comment]
You mean, instead of the symbolic link?

**Comment by mikefaille** [comment]
Yes

**Comment by mikefaille** [comment]
Yes. To point clearly this issue : 

```
ln -s /usr/lib/jvm/java-7-openjdk-amd64 /usr/lib/jvm/jdk 
```

**Comment by davidonlaptop** [comment]
It makes sense.

I guess it should be done for both: java and javac.

**Comment by mikefaille** [comment]
By default, jdk have the priority over jre. You can confirm this using `update-alternatives --display <java>` man documentation : http://manpages.ubuntu.com/manpages/hardy/man8/update-alternatives.8.html

**Comment by codingtony** [comment]
This makes a lot of sense. I was surprised not to see the use of
alternatives in the Dockerfile.

I assumed that the symlink was the way you decided to go with, so I did not
change anything.

I'll fix it with alternatives later today.

**Comment by codingtony** [comment]
BTW alternatives points already to openjdk (it has been done during the installation of the package)

```
> docker run --rm -ti gelog/hadoop  bash

> update-alternatives  --list java
/usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java

> ls -l $(which java)
lrwxrwxrwx 1 root root 22 Sep 22 14:04 /usr/bin/java -> /etc/alternatives/java
```

**Comment by davidonlaptop** [comment]
Great! Does it work for javac too ?

At the beginning we were using Oracle JDK, but had to drop it for licensing
issues. Perhaps that explains the symlink.

-David

**Comment by codingtony** [comment]
yes it does.

```
ls -l $(which javac)
lrwxrwxrwx 1 root root 23 Sep 22 14:13 /usr/bin/javac ->
/etc/alternatives/javac*
```

**Comment by davidonlaptop** [comment]
Perfect!

**Comment by davidonlaptop** [comment]
Fixed by pull request #21 and merged into master.
